tests/integration/tokenomics/tokenomics_example_test.go

```diff
@@ -0,0 +1,95 @@
+package integration_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/cmd/poktrolld/cmd"
+	integration "github.com/pokt-network/poktroll/testutil/integration"
+	testutilproof "github.com/pokt-network/poktroll/testutil/proof"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	tokenomicstypes "github.com/pokt-network/poktroll/x/tokenomics/types"
+)
+
+func init() {
+	cmd.InitSDKConfig()
+}
+
+// This is an example integration test @Olshansk was developing while implementing
+// `testutil/integration/app.go` to test and verify different behaviours from
+// setup, querying, running messages, etc...
+// TODO_TECHDEBT: Delete this once other integration tests exist or refactor
+// it to be something more concrete and useful.Decide if this should be deleted
+func TestTokenomicsIntegrationExample(t *testing.T) {
+	// Create a new integration app
+	integrationApp := integration.NewCompleteIntegrationApp(t)
+
+	// Query and validated the default shared params
+	sharedQueryClient := sharedtypes.NewQueryClient(integrationApp.QueryHelper())
+	sharedQueryParams := sharedtypes.QueryParamsRequest{}
+	sharedQueryResponse, err := sharedQueryClient.Params(integrationApp.SdkCtx(), &sharedQueryParams)
+	require.NoError(t, err)
+	require.NotNil(t, sharedQueryResponse, "unexpected nil queryResponse")
+	require.EqualValues(t, uint64(4), sharedQueryResponse.Params.NumBlocksPerSession)
+
+	// Prepare a request to update the compute_units_to_tokens_multiplier
+	updateTokenomicsParamMsg := &tokenomicstypes.MsgUpdateParam{
+		Authority: integrationApp.Authority(),
+		Name:      "compute_units_to_tokens_multiplier",
+		AsType:    &tokenomicstypes.MsgUpdateParam_AsInt64{AsInt64: 11},
+	}
+
+	// Run the request
+	result := integrationApp.RunMsg(t,
+		updateTokenomicsParamMsg,
+		integration.WithAutomaticFinalizeBlock(),
+		integration.WithAutomaticCommit(),
+	)
+	require.NotNil(t, result, "unexpected nil result")
+
+	// Validate the response is correct and that the value was updated
+	resp := tokenomicstypes.MsgUpdateParamResponse{}
+	err = integrationApp.Codec().Unmarshal(result.Value, &resp)
+	require.NoError(t, err)
+
+	tokenomicsQueryClient := tokenomicstypes.NewQueryClient(integrationApp.QueryHelper())
+
+	tokenomicsQueryParams := tokenomicstypes.QueryParamsRequest{}
+	tokenomicsQueryResponse, err := tokenomicsQueryClient.Params(integrationApp.SdkCtx(), &tokenomicsQueryParams)
+	require.NoError(t, err)
+	require.NotNil(t, tokenomicsQueryResponse, "unexpected nil queryResponse")
+	require.EqualValues(t, uint64(11), uint64(tokenomicsQueryResponse.Params.ComputeUnitsToTokensMultiplier))
+
+	// Commit & finalize the current block, then moving to the next one.
+	integrationApp.NextBlock(t)
```

Comment:
I would've expected this to be mutually exclusive with the usage of the `WithAutomaticFinalizeBlock()` and `WithAutomaticCommit()` options when "running" the message above. I think it would be great if this example demonstrated both but I think a #PUC would be in order in that case.

Reply:
Sorry for the confusion. Let me try to rephrase: It seems to me like calling `integrationApp.NextBlock(t)` here is redundant because no on-chain activity should've happened between calling `integrationApp.RunMsg()` (on line 46) and here (line 67). Since the `#RunMsg()` call includes the finalize and commit options, why do we need to call `#NextBlock()` here?

Reply:
Sorry for the confusion. Let me try to rephrase: It seems to me like calling `integrationApp.NextBlock(t)` here is redundant because no on-chain activity should've happened between calling `integrationApp.RunMsg()` (on line 46) and here (line 67). Since the `#RunMsg()` call includes the finalize and commit options, why do we need to call `#NextBlock()` here?

Reply:
Made some changes so resolving this for now.